rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java

```diff
@@ -198,8 +200,10 @@ private void doCommit() throws SQLException {
     }
 
     private void processLocalCommitWithGlobalLocks() throws SQLException {
-
-        checkLock(context.buildLockKeys());
+        Set<String> lockKeys = context.getLockKeysBuffer();
+        if (CollectionUtils.isNotEmpty(lockKeys)) {
```

Comment:
Is it more appropriate to put logic with lockkey as empty in the checkLock method?

Reply:
ok,i will change

Reply:
![image](https://user-images.githubusercontent.com/45086633/78786050-b84b5e80-79da-11ea-810d-06963a10ca69.png)
the master code